lib/spawn-browser.ts

```diff
@@ -72,14 +79,23 @@ class BrowserProcess implements IBrowserProcess {
   private lastError: Error;
   private hasExited: boolean = false;
 
-  constructor(executablePath: string, args: string[]) {
+  constructor(
+    executablePath: string,
+    args: string[],
+    stdio: "pipe" | "ignore" | "inherit" | null = "inherit",
+  ) {
+    // the types aren't current for this
+    // disabling buffer used to be maxBuffer: null
+    // now it is buffer: false
     const child = execa(executablePath, args, {
-      // disable buffer, pipe to stdout
-      maxBuffer: null as any,
-    });
-    child.stdout.pipe(process.stdout);
-    child.stderr.pipe(process.stderr);
-    child.on("error", (err: Error) => (this.lastError = err));
+      // disable buffer, pipe or drain
+      buffer: false,
+      stdio,
+    } as any);
+    // child is now a thenable for the duration of the promise
+    // ensure we handle its error or we will get an
+    // unhandled rejection warning
+    child.catch(err => (this.lastError = err));
```

Comment:
Would it be useful to keep a list of errors if the child has multiple errors? Not sure if that's even possible.

Reply:
The process exits on error, execa `child` is a promise like for the lifetime of the child process.